sdk/communication/communication-common/test/public/communicationKeyCredentialPolicy.spec.ts

```diff
@@ -58,6 +59,128 @@ describe("CommunicationKeyCredentialPolicy", function () {
+      sendRequest: async (req) => {
+        if (responses.length) {
+          const response = responses.shift()!;
+          response.request = req;
+          return response;
+        }
+        throw new Error("No responses found");
+      },
+    };
+
+    const pipeline = createEmptyPipeline();
+    pipeline.addPolicy(communicationKeyCredentialPolicy);
+    const pipelineResponse = await pipeline.sendRequest(testHttpsClient, pipelineRequest);
+
+    const authHeader = pipelineResponse.request.headers.get("Authorization");
+    const dateHeader = pipelineResponse.request.headers.get("x-ms-date");
+    const hashHeader = pipelineResponse.request.headers.get("x-ms-content-sha256");
+    const hostHeader = pipelineResponse.request.headers.get("Host");
+
+    const authHeaderExpected = await generateAuthHeader(
+      pipelineRequest.method.toLocaleUpperCase(),
+      dateHeader || "",
+      credential.key,
+      "/testPath?testQuery=",
+      "example.com"
+    );
+
+    assert.equal(authHeader, authHeaderExpected);
+    assert.isNotEmpty(dateHeader);
+    assert.isNotEmpty(hashHeader);
+    if (isNode) {
+      assert.isNotEmpty(hostHeader);
+    }
+  });
+});
+
+describe("CommunicationKeyCredentialPolicy", function () {
+  it("signs the request correctly with path and no query", async function () {
+    const credential = new MockKeyCredential("pw==");
+    const communicationKeyCredentialPolicy =
+      createCommunicationAccessKeyCredentialPolicy(credential);
+
+    const pipelineRequest = createPipelineRequest({ url: "https://example.com/testPath" });
+
+    const responses: PipelineResponse[] = [
+      {
+        headers: createHttpHeaders(),
+        request: pipelineRequest,
+        status: 200,
+      },
+    ];
+
+    const testHttpsClient: HttpClient = {
+      sendRequest: async (req) => {
+        if (responses.length) {
+          const response = responses.shift()!;
+          response.request = req;
+          return response;
+        }
+        throw new Error("No responses found");
+      },
+    };
+
+    const pipeline = createEmptyPipeline();
+    pipeline.addPolicy(communicationKeyCredentialPolicy);
+    const pipelineResponse = await pipeline.sendRequest(testHttpsClient, pipelineRequest);
+
+    const authHeader = pipelineResponse.request.headers.get("Authorization");
+    const dateHeader = pipelineResponse.request.headers.get("x-ms-date");
+    const hashHeader = pipelineResponse.request.headers.get("x-ms-content-sha256");
+    const hostHeader = pipelineResponse.request.headers.get("Host");
+
+    const authHeaderExpected = await generateAuthHeader(
+      pipelineRequest.method.toLocaleUpperCase(),
+      dateHeader || "",
+      credential.key,
+      "/testPath",
+      "example.com"
+    );
+
+    assert.equal(authHeader, authHeaderExpected);
+    assert.isNotEmpty(dateHeader);
+    assert.isNotEmpty(hashHeader);
+    if (isNode) {
+      assert.isNotEmpty(hostHeader);
+    }
+  });
```

Comment:
Instead of duplicating code in all 3 test cases, we can extract common functionality, like creating TestClient, Pipeline and Headers (as it done for "generateAuthHeader"). It would become more concise and clear.
Similarly for the assertions as well, like a single method asserting all the headers.

Reply:
Created a new function to do all the core functions as only difference is URL and returned auth header